### PR TITLE
draft of pick_closed_reference_otus

### PR DESCRIPTION
**Pull request by wjurkowski**

### CODE REVIEW

File: tools/qiime/qiime_core/pick_closed_reference_otus.xml [hunk 1]
```diff
@@ -0,0 +1,198 @@
+<tool id="pick_open_reference_otus" name="Perform open-reference OTU picking" version="@WRAPPER_VERSION@.0">
```
Comment (bebatut) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: Can you name it `qiime_pick_closed_reference_otus`?

Comment (bebatut) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: Can you correct the name and the description to fit the Closed-reference, please?

File: tools/qiime/qiime_core/pick_closed_reference_otus.xml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            --input_fp '$input_fp'
+            --output_dir otus
+            #if str( $reference_sequences.reference_sequences_selector ) == 'history'
+                #set $ref = $reference_sequences.history_database
+            #else:
+                #set $data_table = dict([(_[0], _[2]) for _ in $reference_sequences.cached_database.input.options.tool_data_table.data])
+                #set $db = $reference_sequences.cached_database.value
+                #set $ref = $data_table[$db]
+            #end if
+            --reference_fp '$ref'
```
Comment (bebatut) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: What is `$ref` as variable? I can not find it in the `param`

File: tools/qiime/qiime_core/pick_closed_reference_otus.xml [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            #if str( $prefilter.prefilter_refseqs_fp ) == 'yes'
```
Comment (bebatut) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: These variables are not declared as parameters.

File: tools/qiime/qiime_core/pick_closed_reference_otus.xml [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        </conditional>
+        <param argument="--parallel" type="boolean" truevalue="--parallel" falsevalue="" checked="False" label="Run in parallel where available?"/>
+        <param argument="--suppress_taxonomy_assignment" type="boolean" truevalue="--suppress_taxonomy_assignment" falsevalue="" checked="False" label="Skip the taxonomy assignment step?" help="It results in an OTU table without taxonomy"/>
+    </inputs>
+
+    <outputs>
+        <data name="final_otu_map" format="txt" from_work_dir="otus/final_otu_map.txt" label="${tool.name} on ${on_string}: Final OTU map"/>
+        <data name="final_otu_map_mc" format="txt" from_work_dir="otus/final_otu_map_mc*.txt" label="${tool.name} on ${on_string}: Final OTU map without the OTUs failing the minimum specified OTU size"/>
+        <data name="rep_set" format="fasta" from_work_dir="otus/rep_set.fna" label="${tool.name} on ${on_string}: OTU representative sequences"/>
+        <data name="otu_table_mc" format="biom" from_work_dir="otus/otu_table_mc*.biom" label="${tool.name} on ${on_string}: OTU table"/>
```
Comment (bebatut) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: The format is `biom1`

File: tools/qiime/qiime_core/pick_closed_reference_otus.xml [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        </data>
+        <data name="rep_set_tree" format="txt" from_work_dir="otus/rep_set.tre" label="${tool.name} on ${on_string}: Representative set tree">
+            <filter>suppress_align_and_tree is False</filter>
+        </data>
+        <data name="new_refseqs" format="fasta" from_work_dir="otus/new_refseqs.fna" label="${tool.name} on ${on_string}: New reference sequences (OTU + input reference sequences)"/>  
+    </outputs>
+
+    <tests>
+        <test>
+            <param name="input_fps" value="pick_open_reference_otus/sequences.fasta"/>
```
Comment (bebatut) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: Can you create a directory with the test data for this wrapper (in `test-data`) and use specific tests? You can find a list of possible test data for each script in QIIME repo ([here](https://github.com/biocore/qiime/tree/master/qiime_test_data/pick_closed_reference_otus))

File: tools/qiime/qiime_core/pick_closed_reference_otus.xml (hunk 2 above)
Comment (nsoranzo) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: It's defined in Cheetah 2 lines above.

Comment (bebatut) [inline tools/qiime/qiime_core/pick_closed_reference_otus.xml]: Ouuppss :confounded:

**Comment by wjurkowski** [comment]
@bebatut Thanks for comments, Could you explain the difference between the core and add on? It doesn't seem to be function related.

**Comment by bebatut** [comment]
The number of scripts in QIIME is too big to integrate all of them rapidly (in the next months). We decide to define a set of core scripts to focus on and all other integrated scripts are bonus (but will be available also in the ToolShed): we selected the scripts found in the tutorials as core scripts.
But if you think some core scripts are missing, please tell us :)

**Comment by bgruening** [comment]
@wjurkowski thanks for your contribution!

**Comment by wjurkowski** [comment]
We are using routinely these three scripts, therefore at least from our
point of view they could be part of the core:


   -

   Multiple_join_paired_ends
   -

   Multiple_split_libraries_fastq
   - pick_closed_reference_otus

**Comment by bgruening** [comment]
@wjurkowski can you please add the test-data to this PR?

**Comment by bebatut** [comment]
@wjurkowski we are finalizing the QIIME wrappers and we integrated the pick_closed_reference_otus and the others scripts you mentioned in the core set of qiime wrappers. 
We integrated the latter ones and the only remaining script to integrate if the pick_closed_reference_otus. 
Do you want to finish the integration? Or do you mind if I finish it based on what you started? Your contribution will be kept of course :smile:

**Comment by wjurkowski** [comment]
@bebatut I have it on my todo list, but I'm certainly happy if you could help. Yes, please finish it.

**Comment by bebatut** [comment]
@wjurkowski Thanks! I will try to finish it in the next days

**Comment by bebatut** [comment]
I reworked on `pick_closed_reference_otus`, added the test data and test it locally (with planemo)

**Comment by bgruening** [comment]
Great work @wjurkowski and @bebatut! Thanks was the last missing tool of the core package.
I will feature close the qiime wrapper and we can concentrate on fixing bugs until it is green! :)

**Pull request merged**